src/library/scala/collection/generic/IsSeqLike.scala

```diff
@@ -11,34 +13,78 @@ package generic
+    new IsSeqLike[CC0[A0]] {
+      type A = A0
+      type C = View[A]
+      def apply(coll: CC0[A0]): SeqOps[A0, Iterable, View[A0]] = coll
+    }
+
+  implicit def stringIsSeqLike: IsSeqLike[String] { type A = Char; type C = String } =
     new IsSeqLike[String] {
       type A = Char
-      val conversion: String => SeqOps[Char, Seq, String] = s => new SeqOps[Char, Seq, String] {
-        def length: Int = s.length
-        def apply(i: Int): Char = s.charAt(i)
-        def toIterable: Iterable[Char] = new immutable.WrappedString(s)
-        protected[this] def coll: String = s
-        protected[this] def fromSpecificIterable(coll: Iterable[Char]): String = coll.mkString
-        def iterableFactory: IterableFactory[Seq] = Seq
-        protected[this] def newSpecificBuilder(): mutable.Builder[Char, String] = new StringBuilder
-        def iterator(): Iterator[Char] = s.iterator()
-      }
+      type C = String
+      def apply(s: String): SeqOps[Char, Iterable, String] =
+        new SeqOps[Char, Iterable, String] {
+          def length: Int = s.length
+          def apply(i: Int): Char = s.charAt(i)
+          def toIterable: Iterable[Char] = new immutable.WrappedString(s)
+          protected[this] def coll: String = s
+          protected[this] def fromSpecificIterable(coll: Iterable[Char]): String = coll.mkString
+          def iterableFactory: IterableFactory[Iterable] = Iterable
+          protected[this] def newSpecificBuilder(): mutable.Builder[Char, String] = new StringBuilder
+          def iterator(): Iterator[Char] = s.iterator()
+        }
+    }
+
+  implicit def stringViewIsSeqLike: IsSeqLike[StringView] { type A = Char; type C = View[Char] } =
+    new IsSeqLike[StringView] {
+      type A = Char
+      type C = View[Char]
+      def apply(coll: StringView): SeqOps[Char, Iterable, View[Char]] = coll
     }
 
-  implicit def SeqRepr[C[X] <: Seq[X], A0](implicit conv: C[A0] => SeqOps[A0, C, C[A0]]): IsSeqLike[C[A0]] { type A = A0 } =
-    new IsSeqLike[C[A0]] {
+  implicit def arrayIsSeqLike[A0 : ClassTag]: IsSeqLike[Array[A0]] { type A = A0; type C = Array[A0] } =
+    new IsSeqLike[Array[A0]] {
       type A = A0
-      val conversion = conv
+      type C = Array[A0]
+      def apply(a: Array[A0]): SeqOps[A0, Iterable, Array[A0]] =
+        new SeqOps[A, Iterable, Array[A]] {
+          def apply(i: Int): A = a(i)
+          def length: Int = a.length
+          def toIterable: Iterable[A] = mutable.ArraySeq.make(a)
+          protected def coll: Array[A] = a
+          protected def fromSpecificIterable(coll: Iterable[A]): Array[A] = Array.from(coll)
+          def iterableFactory: IterableFactory[Iterable] = Iterable
+          protected def newSpecificBuilder: mutable.Builder[A, Array[A]] = Array.newBuilder
+          def iterator: Iterator[A] = a.iterator
+        }
     }
+
+  implicit def rangIsSeqLike[C0 <: Range]: IsSeqLike[C0] { type A = Int; type C = immutable.IndexedSeq[Int] } =
```

Comment:
Why is this needed / why isn't `seqOpsIsSeqLike` enough?

Reply:
Because `seqOpsIsSeqLike` provides an instance that matches a type constructor `CC0[X]` with one type parameter. `Range` takes no type parameter so it can never be unified with `CC0[X]`.

Reply:
Thanks. Don't we have the same issue with BitSets?

Reply:
Indeed, good catch! I’ve updated the PR.

Reply:
OK, but then also LongMap? Others?

Reply:
Indeed.

But unfortunately a generic definition like the following doesn’t work:

~~~ scala
implicit def iterableIsIterableLike[Repr <: Iterable[A0], A0]: IsIterableLike[Repr] { type A = A0; type C = Iterable[A0 } = …
~~~

Beause scalac always infers `Nothing` for `A0`:

~~~
type arguments [LongMap[String],Nothing] do not conform to method iterableIsIterableLike's type parameter bounds [Repr <: Iterable[A0],A0]
~~~

Maybe there is a trick to get `A0` correctly inferred?

Reply:
I don't know if this is still useful to anyone, but this?
```
implicit def iterableIsIterableLike[Repr <: Iterable[A0], A0]: IsIterableLike[Repr with Iterable[A0]] { type A = A0; type C = Iterable[A0] } = ???
```